Jellyfin.Server/Middleware/BaseUrlRedirectionMiddleware.cs

```diff
@@ -58,9 +58,12 @@ public async Task Invoke(HttpContext httpContext, IServerConfigurationManager se
                     return;
                 }
 
-                if (!startsWithBaseUrl)
+                if (!startsWithBaseUrl
+                    || localPath.Equals(baseUrlPrefix, StringComparison.OrdinalIgnoreCase)
```

Comment:
We already know that localPath starts with baseUrlPrefix
```suggestion
                    || localPath.Length == baseUrlPrefix.Length
```